### PR TITLE
Base model test extend

### PR DESCRIPTION
**Pull request by make-it-git**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | yes
| Fixed issues      | #

## The problem

We need to set some fields on model before saving it during testing

## How I resolved it

Added callback function to verifySave in BaseModelTest

Notify the following people: @darrynten

### CODE REVIEW

File: tests/SageOne/Models/AccountOpeningBalanceModelTest.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ public function testBadImport()
 
     public function testInject()
     {
-        $this->verifyInject(AccountOpeningBalance::class, function ($model, $data) {
+        $this->verifyInject(AccountOpeningBalance::class, function ($model) {
```
Comment (darrynten) [inline tests/SageOne/Models/AccountOpeningBalanceModelTest.php:22]: Is having space between `function` and `()` PSR compliant?

**Review by darrynten — COMMENTED** [review]
I really like what you've done with the callable, I think we will need to leave some documentation in the README and in the example test so that juniors can understand it more

File: tests/SageOne/Models/AccountOpeningBalanceModelTest.php (hunk 1 above)
Comment (make-it-git) [reply tests/SageOne/Models/AccountOpeningBalanceModelTest.php:22]: https://styleci.io/analyses/8PmMDP

Comment (darrynten) [reply tests/SageOne/Models/AccountOpeningBalanceModelTest.php:22]: Perfect! I think it's just been a long day across too many languages :)

**Review by darrynten — APPROVED** [review]
I approve this code ✔️

**Pull request merged**